views/index.html

```diff
@@ -155,21 +164,44 @@
         success: false,
         results: [],
       },
+      genericError: null,
+    },
+    created: function() {
+      const url = new URL(window.location);
+      this.formData.host = url.searchParams.get('host');
+      this.formData.ingressProxyPath = url.searchParams.get('ingressProxyPath');
+      this.formData.cdnProxyPath = url.searchParams.get('cdnProxyPath');
+      this.formData.trafficName = url.searchParams.get('trafficName');
+      this.formData.integrationVersion = url.searchParams.get('integrationVersion');
     },
     methods: {
       handleSubmit() {
         this.testResult.status = 'pending';
         this.testResult.results = [];
+        this.genericError = null;
         const vm = this;
         fetch('/api/test/run-tests', {
           method: 'POST',
           headers: {
             'Content-Type': 'application/json',
           },
           body: JSON.stringify({ ...this.formData }),
-        }).then(res => res.json()).then(res => {
+        }).then(res => {
```

Comment:
Let's refactor this whole function to async/await to make it more readable.

Reply:
Done 👍